atari-DQN/train_atari_env.py

```diff
@@ -0,0 +1,225 @@
+                               device=device,
+                               summary_writer=None)  # TODO implement summary writer
+    # TODO: implement memory buffer location
+
+    # =====================================================
+    # Start interacting with environment
+    for episode_idx in range(args.num_episode):
+
+        agent.begin_episode()
+        observation = environment.reset()
+
+        cumulative_reward = 0.0
+        steps = 0
+
+        while True:
+
+            action = agent.step(observation)
+            observation, reward, done, info = environment.step(action)
+            # TODO: see if reward is clipped
+
+            agent.store_transition(action, observation, reward, done)
+
+            # Tracker variables
+            cumulative_reward += reward
+            steps += 1
+
+            if done:
+                # =========================================
+                # Logging stuff
+                # Compute logging variables
+                avg_policy_net_loss = 0.0
+                if agent.episode_total_policy_loss > 0.0:
+                    avg_policy_net_loss = agent.episode_total_policy_loss / \
+                                          agent.episode_total_optim_steps
+                # TODO: might be nice to compute the final epsilon per episode
+
+                logtuple = LogTupStruct(episode_idx=episode_idx, steps=steps,
+                                        buffer_size=len(agent.memory),
+                                        training_steps=agent.training_steps,
+                                        returns=cumulative_reward,
+                                        policy_net_loss=avg_policy_net_loss)
+
+                # Write log
+                log_str = '||'.join([str(e) for e in logtuple])
+                if args.log_path is not None:
+                    logger.info(log_str)
+                else:
+                    print(log_str)
+
+                # # =========================================
+                # Break out of current episode
+                break
+
+
+if __name__ == "__main__":
+
+    # TODO: have a hyperparmeter .config file for the future
+
+    # =====================================================
+    # Initialize the argument parser
+    parser = argparse.ArgumentParser(description='DQN for atari environment')
+
+    # Environmental parameters
+    parser.add_argument('--env-name', type=str, default='MsPacman-v0', metavar='N',
+                        help='environment to initialize (default: MsPacman-v0)')
+    parser.add_argument('--num-episode', type=int, default=500, metavar='N',
+                        help='number of episodes to run the environment for (default: 500)')
+    parser.add_argument('--frame-skips', type=int, default=1, metavar='N',
+                        help="""number of frames to repeat each action for, the agent does
+                                not observe the in-between frames (default: 1)""")
```

Comment:
Change this as setting `frame_skips = 1` results in the AtariPreprocessing's max-pooling step breaking down (just returns the first frame constantly). Should be something like: 
```python
parser.add_argument('--frame-skips', type=int, default=4, metavar='N',
                        help="""number of frames to repeat each action for, the agent does
                                not observe the in-between frames. Note that if set to 1 the
                                AtariPreprocessing max pooling breaks down(default: 4)""")
```